libs/ledger-live-common/src/exchange/swap/hooks/live-app-migration/useSwapLiveConfig.ts

```diff
@@ -1,14 +1,38 @@
+import { Feature_PtxSwapCoreExperiment } from "@ledgerhq/types-live/lib/feature";
 import { useFeature } from "../../../../featureFlags";
 
+// check if the variant is valid for core rollout experiment
+export type CoreExperimentParams = NonNullable<Feature_PtxSwapCoreExperiment["params"]>;
+export type ValidVariant = CoreExperimentParams["variant"];
+
 // used to enable the Swap Live App globally
 export function useSwapLiveConfig() {
-  const demoZero = useFeature("ptxSwapLiveAppDemoZero");
-  const demoOne = useFeature("ptxSwapLiveAppDemoOne");
   const demoThree = useFeature("ptxSwapLiveAppDemoThree");
+  const demoOne = useFeature("ptxSwapLiveAppDemoOne");
+  const demoZero = useFeature("ptxSwapLiveAppDemoZero");
+  const coreExperiment = useFeature("ptxSwapCoreExperiment");
+
+  const validVariants: readonly ValidVariant[] = ["Demo0", "Demo3", "Demo3Thorswap"] as const;
```

Comment:
Can you replace this by what you used on LLD so that it's dynamic ?
```
export type CoreExperimentParams = NonNullable<Feature_PtxSwapCoreExperiment["params"]>;
export type ValidVariant = CoreExperimentParams["variant"];
```